src/stage_model.py

```diff
@@ -41,19 +41,37 @@ def add_tokenizer(artifact_dir):
     tokenizer = AutoTokenizer.from_pretrained("answerdotai/ModernBERT-base")
     tokenizer.save_pretrained(artifact_dir)
 
+def add_custom_inference_script(artifact_dir):
+    """Add custom inference script to the artifact directory.
+    Args:
+        artifact_dir (str): Path to the artifact directory.
+    """
+    code_dir = os.path.join(artifact_dir, "code")
+    os.makedirs(code_dir, exist_ok=True)
+
+    # Read the contents of your existing inference.py
```

Comment:
This function is straightforward enough to understand that the inline comments can be safely removed.